ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala

```diff
@@ -120,38 +122,65 @@ class Endpoints(
                 )
               )
           }
-        } yield response
+        } yield transformedResponse
       }
     }
 
   private def logJsonRequestAndResult(implicit
       lc: LoggingContextOf[InstanceUUID with RequestID]
   ): Directive0 = {
-    def logWithHttpMessageBodyIfAvailable(
-        httpMessage: HttpMessage,
+    def logWithHttpMessageBodyIfAvailable[A <: HttpMessage](
+        httpMessage: A,
         msg: String,
-        bodyKind: String,
-    ): Future[Unit] =
+        kind: String,
+    ): HttpMessage =
       if (
         httpMessage
           .header[`Content-Type`]
           .map(_.contentType)
           .contains(ContentTypes.`application/json`)
-      )
-        httpMessage
-          .entity()
-          .toStrict(maxTimeToCollectRequest)
-          .map(it =>
+      ) {
+        httpMessage.entity.contentLengthOption match {
+          // Limit logging of bodies to content with size of less than 10 KiB
+          case Some(length) if length < 1024 * 10 =>
```

Comment:
I would recommend extracting this magical number into its own variable and giving it a name, e.g. `HttpRequestBodyLoggingSizeLimit`.

Reply:
Yeah good idea, will add minimal explanation why I chose 10 KiB too.